### PR TITLE
Fix delete and find by methods

### PR DESCRIPTION
**Pull request by nicsnet**
`delete` now supports deletion of records consisting of a composite primary key schema.
`find_by` and `find` methods now also support querying for records with a composite primary key schema.

@pixelpogo @dakeyras7 @mAu888 @s-ka @inf0rmer please review

### CODE REVIEW

File: spec/dynameister/finders_spec.rb [hunk 1]
```diff
@@ -36,6 +37,25 @@
...
+
+      let(:book) { Book.create(uuid: "id", rank: 42, author_id: 1) }
+
+      subject { Book.find_by(key: { uuid: book.uuid, rank: book.rank }) }
+
+      it "returns a book with the given primary key pair" do
+        expect(subject.uuid).to eq "id"
+      end
+
+      it "returns an instance of document" do
```
Comment (baschtl) [inline spec/dynameister/finders_spec.rb]: Shouldn't it be `"returns an instance of book"`?

File: spec/dynameister/persistence_spec.rb [hunk 2]
```diff
@@ -106,18 +107,38 @@
...
+
+      after { Language.delete_table }
+
+      let!(:language) { Language.create(locale: "GER", displayable: true, rank: 42) }
 
-    let!(:language) { Language.create(locale: "GER", displayable: true, rank: 42) }
+      subject! { language.delete }
 
-    subject!        { language.delete }
+      # TODO: replace query with find_by
```
Comment (baschtl) [inline spec/dynameister/persistence_spec.rb]: Is it possible to exchange query?

**Comment by baschtl** [comment]
@nicsnet Nice. Now go and fix Codeclimate. :P

File: lib/dynameister/finders.rb [hunk 3]
```diff
@@ -8,21 +8,16 @@ module Finders
...
 
-      def find(*hash_keys)
-        hash_keys = Array(hash_keys.flatten.uniq)
-        if hash_keys.count == 1
-          find_by(hash_key: { hash_key.name => hash_keys.first })
-        else
-          options = { hash_key.name => hash_keys }
-          all(options)
-        end
+        keys = { hash_key.name => hash }
```
Comment (inf0rmer) [inline lib/dynameister/finders.rb]: IMO you could use the same `tap`trick as in `fields.rb#primary_key`, to keep the code consistent.

**Comment by inf0rmer** [comment]
:+1: from my side, nice work!

**Comment by pixelpogo** [comment]
:+1:  from my side as well. But only if you add the Codeclimate badge to the Readme... ;-)

**Pull request merged**